### PR TITLE
Remove forbidden `@author` tag from import/export modules

### LINKED ISSUE #36338: [Issue] Remove forbidden `@author` tag from import/export modules
This issue is automatically created based on existing pull request: magento/magento2#36304: Remove forbidden `@author` tag from import/export modules

---------
### Description
According to https://devdocs.magento.com/guides/v2.4/coding-standards/docblock-standard-general.html#documentation-space, the `@author` tag is not permitted in Magento. This pull request removes this tag from some modules. Given there are so many instances of this tag, I've opened a small pull request to get the process started. I expect that the linter will force me to fix several other coding standards violations on the way, so having a smaller pull request means that task is easier to manage. I plan to open more pull requests to tackle the other instances of this tag.

See also https://github.com/magento/magento-coding-standard/pull/382 and https://github.com/magento/magento-coding-standard/issues/167

### Manual testing scenarios
There are not code changes in this pull request. This pull request only removes forbidden comments.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by fredden**
### Description
According to https://devdocs.magento.com/guides/v2.4/coding-standards/docblock-standard-general.html#documentation-space, the `@author` tag is not permitted in Magento. This pull request removes this tag from some modules. Given there are so many instances of this tag, I've opened a small pull request to get the process started. I expect that the linter will force me to fix several other coding standards violations on the way, so having a smaller pull request means that task is easier to manage. I plan to open more pull requests to tackle the other instances of this tag.

See also https://github.com/magento/magento-coding-standard/pull/382 and https://github.com/magento/magento-coding-standard/issues/167

### Manual testing scenarios
There are not code changes in this pull request. This pull request only removes forbidden comments.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#36338: Remove forbidden `@author` tag from import/export modules

### CODE REVIEW

**Comment by fredden** [comment]
@magento run all tests

**Comment by fredden** [comment]
@magento run all tests

File: app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php [hunk 1]
```diff
@@ -568,10 +567,10 @@ protected function getTierPrices(array $listSku, $table)
                 if (isset($price[0]) && !empty($price[0]) || isset($price[1]) && !empty($price[1])) {
                     $select->orWhere('ap.percentage_value IS NOT NULL');
                 }
-                if (isset($updatedAtFrom) && !empty($updatedAtFrom)) {
```
Comment (antoninobonumore) [inline app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php:571]: This seems out of scope from the PR.

**Review by antoninobonumore — CHANGES_REQUESTED** [review]

File: app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php (hunk 1 above)
Comment (fredden) [reply app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php:571]: @antoninobonumore I agree. The way that Adobe have set up GitHub means that any files being modified must 100% pass the current coding standard(s). This change was required to appease phpstan. See the failing test here: https://github.com/magento/magento2/runs/8897630924

```
/var/www/html/dev/tests/static/testsuite/Magento/Test/Php/LiveCodeTest.php:257

2) Magento\Test\Php\LiveCodeTest::testPhpStan
PHPStan detected violation(s):
 ------ ----------------------------------------------------------------------- 
  Line   app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPri  
         cing.php                                                               
 ------ ----------------------------------------------------------------------- 
  570    Variable $updatedAtFrom in empty() always exists and is not falsy.     
  573    Variable $updatedAtTo in empty() always exists and is not falsy.       
 ------ ----------------------------------------------------------------------- 


 [ERROR] Found 2 errors    
```

Comment (fredden) [reply app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php:571]: This is an example of why I didn't want to remove all the `@author` tags in one go.

Comment (antoninobonumore) [reply app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php:571]: Thanks for clarifying this @fredden 
This context makes this change acceptable then.

**Review by antoninobonumore — APPROVED** [review]

File: app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php (hunk 1 above)
Comment (ihor-sviziev) [reply app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php:571]: TBH I would revert this change and add ignore to avoid breaking logic behind it.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
https://github.com/magento/magento2/pull/36304/files#r996760769

File: app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php (hunk 1 above)
Comment (fredden) [reply app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php:571]: The logic isn't broken by this change. The complaint from phpstan is valid. These variables either are undefined (because they're declared inside an `if` statement further up the file) or are a string from `$this->_localeDate->date(...)->format('Y-m-d H:i:s');`, so the `isset()` check makes sense, but the `empty()` is unnecessary. The `empty()` check for `$price[...]` makes sense though, as that may be an "empty" value (like `null`).

@ihor-sviziev would you prefer that I open a separate pull request to fix this complaint from phpstan?

Comment (ihor-sviziev) [reply app/code/Magento/AdvancedPricingImportExport/Model/Export/AdvancedPricing.php:571]: Make sense! Thx for explanation

**Review by ihor-sviziev — APPROVED** [review]

**Comment by fredden** [comment]
@magento run Functional Tests CE

**Comment by engcom-Lima** [comment]
@magento give me test instance

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

As per [document](https://developer.adobe.com/commerce/php/coding-standards/docblock/#inheritdoc-tag) @author tag is not permitted in Magento. Since, this pull request only removes forbidden comments. Hence , tested below scenarios after applying PR changes:

**Manual Scenarios:**

1. Place order as guest / customer from frontend.
2. Added product to wishlist.then, moving to place an order.
3. Done review for a product.
4. From admin side, created invoice , shipped, credit memo.
5. Approve the review from admin side.
6. Created email template.
7. Created simple product.

**Comment by engcom-Lima** [comment]
@magento create issue
